DQMServices/FwkIO/plugins/DQMRootSource.cc

```diff
@@ -533,12 +551,20 @@ std::shared_ptr<edm::FileBlock> DQMRootSource::readFile_() {
     if (!isGoodFile && m_skipBadFiles)
       continue;
 
-    m_openFiles.insert(m_openFiles.begin(), file);
+    std::unique_ptr<std::string> guid{file->Get<std::string>(kCmsGuid)};
+    if (not guid) {
+      guid = std::make_unique<std::string>();
```

Comment:
I was actually thinking using the UUID of `TFile`, e.g. along
```suggestion
      guid = std::make_unique<std::string>(file->GetUUID().AsString());
      std::transform(guid->begin(), guid->end(), guid.begin(), (int (*)(int))std::toupper);
```
as it was before #28622.

Reply:
seems reasonable